mlflow/tracing/clients/__init__.py

```diff
@@ -11,4 +12,8 @@
 
 
 def get_trace_client() -> TraceClient:
-    return InMemoryTraceClientWithTracking.get_instance()
+    """Get the trace client instance based on the environment."""
+    if is_in_databricks_runtime():
+        return InMemoryTraceClientWithTracking.get_instance()
```

Comment:
What about remote workflows that use `mlflow.set_tracking_uri("databricks")` and `mlflow.login()`? We need to support those.

Reply:
True, will be it be enough to check tracking URI is "databricks"? (not scoping OSS tracking server now)

Reply:
We can use `is_databricks_uri`: https://github.com/harupy/mlflow/blob/74fbc2440f946ee3f9665d1be461b85e6fd4e048/mlflow/utils/uri.py#L81

Reply:
If we support tracing in OSS, will we remove if and always return `InMemoryTraceClientWithTracking`?

Reply:
I think it depends on how we store traces in OSS. In Databricks we store traces in Run table but it might not be the best option for OSS (given that the DB or filesystem that stores runs are exposed to users). Eventually it might be the same client and everything is abstracted behind the tracking server, but there may be subtle difference.